src/main/fc/init.c

```diff
@@ -259,26 +259,6 @@ void sdCardAndFSInit()
 
 void init(void)
 {
-#ifdef USE_HAL_DRIVER
```

Comment:
What is the reason to push this into separate files (and make maintenance harder)?

Reply:
the initialisation order was incorrect.  HAL_Delay() was being used in early startup code before init() was called.  this fixes it. see SystemClock_Config and commit message.